pop_tools/budget.py

```diff
@@ -0,0 +1,328 @@
+import datetime
+import glob
+import warnings
+
+import numpy as np
+import xarray as xr
+
+from .grid import get_grid
+
+NMOLS_TO_MOLYR = 1e-9 * 86400.0 * 365.0
+
+
+def _process_grid(grid):
+    """Load in POP grid and process coordinates required for budget calculation."""
+    grid_f = get_grid(grid)
+    dz = grid_f.dz.drop('z_t')
+    area = (grid_f.TAREA).rename('area')
+    vol = (area * dz).rename('vol')
+    z = (grid_f.z_w_bot.drop('z_w_bot')).rename('z')
+    mask = grid_f.REGION_MASK.rename('mask')
+    return xr.merge([area, vol, z, mask])
+
+
+def _compute_kmax(budget_depth, z):
+    """Compute the k-index of the maximum budget depth"""
+    # NOTE: Change this to other argmin() method.
+    kmax = (z <= budget_depth).argmin() - 1
+    kmax = kmax.values
+    # This only occurs when the full depth is requested (or `budget_depth` is greater than the maximum POP depth`)
+    if kmax == -1:
+        kmax = None
+    return kmax
+
+
+def _convert_to_tendency(ds, normalizer, sign=1, kmax=None):
+    """Converts from volume- and area-dependent units to tendencies"""
```

Comment:
@bradyrx, some tiny comments, but in general things here look clean to me. It would be nice to update the docstrings for better **function signatures** documentation.  In some places, it's hard to tell parameter types without reading the entire function.

Reply:
Thanks for bringing this up @andersys005. I didn't know if it was conventional to document private functions extensively or not. But it will be good for any future development on this (particularly when it isn't me doing it). I'll update docstrings in this next round of commits.

Reply:
>  I'll update docstrings in this next round of commits.

Thank you for doing this

> But it will be good for any future development on this (particularly when it isn't me doing it).

I concur! Enough documentation saves many hours of explanation. It takes less time for new contributors to come into the project.

Reply:
@bradyrx,

>  I didn't know if it was conventional to document private functions extensively or not.

Good point. I realized that I didn't know whether there exists a common practice for this :)  

Upon this very topic, [PEP 8](https://www.python.org/dev/peps/pep-0008/) says:

> Write docstrings for all public modules, functions, classes, and methods. Docstrings are not necessary for non-public methods, but you should have a comment that describes what the method does. This comment should appear after the "def" line.


So, feel free to update the docstrings for private functions as you see fit.